File: exercise-scala/exercise-scala-basic/src/test/scala/me/shu/exercise/scala/basic/LinkedHashMapTest.scala
package me.shu.exercise.scala.basic

import org.scalatest.{Matchers, FlatSpec}

import scala.collection.JavaConversions._


class LinkedHashMapTest extends FlatSpec with Matchers {

  "LinkedHashMap" should "pop values in last-in-first-out order" in {

    val cacheSize = 5
    val lruCache = new java.util.LinkedHashMap[Integer, Integer](10, 0.75f, true) {
      protected override def removeEldestEntry(eldest: java.util.Map.Entry[Integer, Integer]): Boolean = {
        return this.size > cacheSize
      }
    }

    (0 until 3) foreach { i => lruCache.put(i, i) }
    lruCache.get(0)
    (3 until 6) foreach { i => lruCache.put(i, i) }

    lruCache.foreach(println)

  }

}

File: exercise-java/exercise-java-basic/src/test/java/me/shu/exercise/java/basic/LinkedHashMapTest.java
package me.shu.exercise.java.basic;


import org.junit.Test;

import java.util.LinkedHashMap;
import java.util.Map;

public class LinkedHashMapTest {

    @Test
    public void basicTest() {

        final int cacheSize = 5;

        LinkedHashMap<Integer, Integer> lruCache = new LinkedHashMap<Integer, Integer>(10, 0.8f, true) {
            @Override
            protected boolean removeEldestEntry(Map.Entry<Integer, Integer> eldest) {
                return this.size() > cacheSize;
            }
        };


        for (int i = 0; i < 5; i++) {
            lruCache.put(i, i);
        }

        for (Map.Entry entry : lruCache.entrySet()) {
            System.out.println(entry.getValue());
        }

    }
}
